packages/permission-controller/src/PermissionController.ts

```diff
@@ -1638,7 +1664,7 @@ export class PermissionController<
     }
 
     this.setValidatedPermissions(origin, permissions);
-    return permissions;
+    return { permissions, data };
```

Comment:
I'm afraid that changing `grantPermissions` return type will break all the implementations outside of the controller, we might want to avoid this ?

Reply:
Yeah, I looked into this before changing the type. The `SnapsController` is the only place using `grantPermissions`, so it's not breaking anywhere else. The rest of the extension uses the `requestPermissions` entry point.